### PR TITLE
Fix: extension crash on Fennec due to missing icon APIs in chrome.bro…

### LINKED ISSUE #13035: 2017.10.4 doesnt work on the last Firefox for android
2017.10.4 doesnt work on the last Firefox for android

### PR DESCRIPTION
**Pull request by Hainish**
…wserAction (fixes #13035)

### CODE REVIEW

**Comment by Hainish** [comment]
Ready for review @cowlicks

File: chromium/background.js [hunk 1]
```diff
@@ -130,11 +130,13 @@ function updateState () {
     iconState = 'blocking';
   }
 
-  chrome.browserAction.setIcon({
-    path: {
-      38: 'icons/icon-' + iconState + '-38.png'
-    }
-  });
+  if ('setIcon' in chrome.browserAction) {
```
Comment (koops76) [inline chromium/background.js]: `if (chrome.browserAction.setIcon) {` would work just as well and is more readable.

File: chromium/background.js [hunk 2]
```diff
@@ -147,11 +149,15 @@ function updateState () {
     const tabId = tabs[0].id;
     const activeCount = getActiveRulesetCount(tabId);
 
-    chrome.browserAction.setBadgeBackgroundColor({ color: '#666666', tabId });
+    if ('setBadgeBackgroundColor' in chrome.browserAction) {
```
Comment (koops76) [inline chromium/background.js]: `if (chrome.browserAction.setBadgeBackgroundColor) {`

File: chromium/background.js [hunk 3]
```diff
@@ -147,11 +149,15 @@ function updateState () {
...
+      chrome.browserAction.setBadgeBackgroundColor({ color: '#666666', tabId });
+    }
 
     const showBadge = activeCount > 0 && isExtensionEnabled && showCounter;
 
-    chrome.browserAction.setBadgeText({ text: showBadge ? String(activeCount) : '', tabId });
+    if ('setBadgeText' in chrome.browserAction) {
```
Comment (koops76) [inline chromium/background.js]: `if (chrome.browserAction.setBadgeText) {`

File: chromium/background.js (hunk 1 above)
Comment (Hainish) [reply chromium/background.js]: I actually think this is more readable.

Comment (koops76) [reply chromium/background.js]: It's a bit harder to read in my opinion. It's common in JS to use such property existence checks.

Comment (Hainish) [reply chromium/background.js]: Even if so, I like the explicit method you're searching for separated, as a syntactic nicety.

**Pull request merged**